src/network/networkd-netdev-wireguard.c

```diff
@@ -0,0 +1,625 @@
+/***
+    This file is part of systemd.
+
+    Copyright 2016 Jörg Thalheim <joerg@higgsboson.tk>
+
+    systemd is free software; you can redistribute it and/or modify it
+    under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.
+
+    systemd is distributed in the hope that it will be useful, but
+    WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <sys/ioctl.h>
+#include <net/if.h>
+
+#include "alloc-util.h"
+#include "parse-util.h"
+#include "in-addr-util.h"
+#include "fd-util.h"
+#include "strv.h"
+#include "hexdecoct.h"
+#include "string-util.h"
+#include "networkd-netdev-wireguard.h"
+#include "networkd-link.h"
+#include "networkd.h"
+
+#define WG_SET_DEVICE (SIOCDEVPRIVATE + 1)
+#define MAX_CONCURRENT_QUERIES 2048
+
+static void resolve_endpoints(NetDev *netdev);
+
+static WireguardPeer *new_peer(Wireguard *w, unsigned section) {
```

Comment:
hmm, to follow our usualy coding style, please name this "wireguard_peer_new()"...